src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -144,52 +215,79 @@ sealed trait LazyModuleImpLike extends RawModule
   val auto: AutoBundle
   protected[diplomacy] val dangles: Seq[Dangle]
 
-  // .module had better not be accessed while LazyModules are still being built!
-  require (!LazyModule.scope.isDefined, s"${wrapper.name}.module was constructed before LazyModule() was run on ${LazyModule.scope.get.name}")
+  /** [[wrapper.module]] had better not be accessed while LazyModules are still being built! */
+  require (LazyModule.scope.isEmpty, s"${wrapper.name}.module was constructed before LazyModule() was run on ${LazyModule.scope.get.name}")
 
+  /** set name from [[wrapper]]. */
   override def desiredName = wrapper.desiredName
   suggestName(wrapper.suggestedName)
 
-  implicit val p = wrapper.p
+  /** [[Parameters]] for diplomatic [[Module]]s. */
+  implicit val p: Parameters = wrapper.p
 
+  /** [[instantiate]] will be called when a instance of [[LazyModuleImp]] is created. */
   protected[diplomacy] def instantiate() = {
     val childDangles = wrapper.children.reverse.flatMap { c =>
       implicit val sourceInfo = c.info
+      /** calling [[c.module]] will push the real [[Module]] into [[chisel3.internal.Builder]],
+        * notice this place is not calling [[instantiate]], it just push the [[c.module]] to Builder.
+        * */
       val mod = Module(c.module)
+      /** ask each child to finish instantiate. */
       mod.finishInstantiate()
+      /** return dangles of each child. */
```

Comment:
what is a dangles...

Reply:
changed to
```
      /** return [[Dangle]]s of each child. */
```
`Dangle` has been documented